Added toggler component

Ali-Khamis opened the pull request:
Issue : [Creating toggler issue link](https://github.com/m3ntorship/pickify-frontend/issues/12)

#Description 
-Added button component with storybook and its tests

EmanFouad commented on modules/shared/components/toggler/index.tsx:
```diff
@@ -0,0 +1,58 @@
+    >
+      <div className={circleClasses} />
+    </button>
+  );
```
It's better to use input type="checkbox" instead of button because it include important attributes like checked, so replace it and upon it there will be some changes in style and testing.

Ali-Khamis commented:
#Changes done after requested from the reviewer 
- Changed the button to input (check box)
- Updated the tests
- Updated the types
- Added a translate to tailwind config as it's needed for the toggle

EmanFouad commented:
Thanks for the clear definition of component's props, it's helpful,
**please consider these following small changes in style:**
size: default & small:
- checked & disabled  ----> opacity-50 not opacity-25
- unchecked & disabled -------> bg-grey-shd6 not opacity-25
![1](https://user-images.githubusercontent.com/14257006/118378982-9d80eb80-b5d7-11eb-8840-169f23198883.PNG)
![2](https://user-images.githubusercontent.com/14257006/118378983-9fe34580-b5d7-11eb-8c4d-8e96d34a7a22.PNG)


**at testing with RTL :**
you can add one more test for testing disabled case

Ali-Khamis commented:
Changes requested done

EmanFouad reviewed (approved):
LGTM
